README.md

```diff
@@ -1,2 +1,243 @@
+
+## Overview
+
+This repository serves as:
+- An **abstraction layer** for other libraries implementing specific RTOS functionality
+- A **common interface** for modules that need OS services without being tied to a specific RTOS
+- A **standardized API** for embedded systems targeting various RTOS platforms (FreeRTOS, ThreadX, Zephyr, etc.)
+
+## Features
+
+DMOSI provides weak implementations (stubs) for the following operating system primitives:
+
+### 1. **Mutex API**
+Thread synchronization through mutual exclusion locks:
+- `dmosi_mutex_create()` - Create a mutex (recursive or non-recursive)
+- `dmosi_mutex_destroy()` - Destroy a mutex
+- `dmosi_mutex_lock()` - Lock a mutex
+- `dmosi_mutex_unlock()` - Unlock a mutex
+
+### 2. **Semaphore API**
+Counting semaphores for resource management:
+- `dmosi_semaphore_create()` - Create a semaphore with initial and max counts
+- `dmosi_semaphore_destroy()` - Destroy a semaphore
+- `dmosi_semaphore_wait()` - Wait on a semaphore (with timeout)
+- `dmosi_semaphore_post()` - Post to a semaphore
+
+### 3. **Thread API**
+Thread creation and management:
+- `dmosi_thread_create()` - Create a new thread
+- `dmosi_thread_destroy()` - Destroy a thread
+- `dmosi_thread_join()` - Wait for thread completion
+- `dmosi_thread_current()` - Get current thread handle
+- `dmosi_thread_sleep()` - Sleep for specified milliseconds
+
+### 4. **Process API**
+Process-level operations (for RTOS that support processes):
+- `dmosi_process_create()` - Create a new process
+- `dmosi_process_destroy()` - Destroy a process
+- `dmosi_process_current()` - Get current process handle
+
+### 5. **Queue API**
+Inter-task message queues:
+- `dmosi_queue_create()` - Create a queue with specified item size and length
+- `dmosi_queue_destroy()` - Destroy a queue
+- `dmosi_queue_send()` - Send data to queue (with timeout)
+- `dmosi_queue_receive()` - Receive data from queue (with timeout)
+
+### 6. **Timer API**
+Software timers for periodic or one-shot callbacks:
+- `dmosi_timer_create()` - Create a timer with callback
+- `dmosi_timer_destroy()` - Destroy a timer
+- `dmosi_timer_start()` - Start a timer
+- `dmosi_timer_stop()` - Stop a timer
+- `dmosi_timer_reset()` - Reset a timer
+
+## Usage
+
+### Basic Integration
+
+Add DMOSI to your CMake project using FetchContent:
+
+```cmake
+include(FetchContent)
+
+FetchContent_Declare(
+    dmosi
+    GIT_REPOSITORY https://github.com/choco-technologies/dmosi.git
+    GIT_TAG        main
+)
+
+FetchContent_MakeAvailable(dmosi)
+
+target_link_libraries(your_target PRIVATE dmosi)
+```
+
+### Implementing RTOS-Specific Functions
+
+To implement DMOSI for your specific RTOS, override the weak implementations by providing strong implementations in your library:
+
+```c
+// Example: FreeRTOS implementation
+#include "dmosi.h"
+#include "FreeRTOS.h"
```

Comment:
Nie masz załączać żadnego freertos ani innych systemów. To ma być tylko repozytorium z API, bez implementacji